sklearn/linear_model/tests/test_quantile.py

```diff
@@ -0,0 +1,102 @@
+    # Test that model estimates percentage of points that lie below the prediction
+    X, y = make_regression(n_samples=1000, n_features=20, random_state=0, noise=1.0)
+    for q in [0.5, 0.9, 0.05]:
+        quant = QuantileRegressor(quantile=q, alpha=0).fit(X, y)
+        fraction_below = np.mean(y < quant.predict(X))
+        assert_almost_equal(fraction_below, q, 2)
+
+
+def test_quantile_is_approximately_sparse():
+    # now most of coefficients are not exact zero, but with large n_samples they are close enough
+    X, y = make_regression(n_samples=3000, n_features=100, n_informative=10, random_state=0, noise=1.0)
+    q = QuantileRegressor(l1_ratio=1, alpha=0.1).fit(X, y)
+    share_zeros = np.mean(np.abs(q.coef_) > 1e-1)
+    assert_almost_equal(share_zeros, 0.1, 2)
+
+
+def test_quantile_without_intercept():
+    X, y = make_regression(n_samples=300, n_features=20, random_state=0, noise=1.0)
+    quant = QuantileRegressor(alpha=1e-4, fit_intercept=False).fit(X, y)
+    # check that result is similar to Huber
+    huber = HuberRegressor(epsilon=1 + 1e-4, alpha=1e-4, fit_intercept=False).fit(X, y)
+    assert_almost_equal(huber.intercept_, quant.intercept_, 1)
+    assert_almost_equal(huber.coef_, quant.coef_, 1)
+    # check that we still predict fraction
+    fraction_below = np.mean(y < quant.predict(X))
+    assert_almost_equal(fraction_below, 0.5, 1)
+
+
+def test_quantile_sample_weight():
+    # test that with unequal sample weights we still estimate weighted fraction
+    n = 500
+    X, y = make_regression(n_samples=n, n_features=10, random_state=0, noise=10.0)
+    weight = np.ones(n)
+    # when we increase weight of upper observaions, estimate of quantile should go up
+    weight[y > y.mean()] = 100
+    quant = QuantileRegressor(quantile=0.5, alpha=1e-4).fit(X, y, sample_weight=weight)
+    fraction_below = np.mean(y < quant.predict(X))
+    assert_greater(fraction_below, 0.5)
+    weighted_fraction_below = np.sum((y < quant.predict(X)) * weight) / np.sum(weight)
+    assert_almost_equal(weighted_fraction_below, 0.5, 2)
+
+
+def test_quantile_incorrect_quantile():
+    X, y = make_regression(n_samples=10, n_features=1, random_state=0, noise=1)
+    with assert_raises(ValueError):
+        quant = QuantileRegressor(quantile=2.0).fit(X, y)
+    with assert_raises(ValueError):
+        quant = QuantileRegressor(quantile=1.0).fit(X, y)
+    with assert_raises(ValueError):
+        quant = QuantileRegressor(quantile=0.0).fit(X, y)
+
+
+def test_quantile_warm_start():
+    X, y = make_regression()
+    warm = QuantileRegressor(
+        fit_intercept=True, alpha=1.0, max_iter=10000, warm_start=True, gtol=1e-1)
+    warm.fit(X, y)
+    warm_coef = warm.coef_.copy()
+    warm.fit(X, y)
+
+    # SciPy performs the tol check after doing the coef updates, so
+    # these would be almost same but not equal.
+    assert_array_almost_equal(warm.coef_, warm_coef, 1)
+
+
+def test_quantile_convergence():
```

Comment:
This test is failing.

Reply:
Applied a larger dataset (1000 obs vs 500 before). The convergence improved just enough for the test to pass.